### ISSUE TITLE
test_call_straylight1 intermittently fails

### ISSUE THREAD

**Issue by stscijgbot** [spacetelescope/jwst#4056]
Issue [JP-1015](https://jira.stsci.edu/browse/JP-1015) was created by James Davies:

If one runs the `jwst` unit tests using `pytest-xdist`, the failure occasionally happens:
```
__________________________________________________________________ test_call_straylight1 ___________________________________________________________________
[gw0] darwin -- Python 3.7.4 /Users/jdavies/miniconda3/envs/jwst_yaml/bin/python

_jail = None, miri_mrs = <IFUImageModel(20, 20) from test_miri.fits>

    def test_call_straylight1(_jail, miri_mrs):
        """ test possible user options are set up correctly """
    
        collect_pipeline_cfgs('./config')
    
        # Test the ModShepard power is in the correct range
        # set the power to outside the upper range
        step = StraylightStep.from_config_file('config/straylight.cfg')
        step.override_straylight = 'dummy.asdf'
        step.method = 'ModShepard'
        step.power = 6
>       result = step.run(miri_mrs)

/Users/jdavies/dev/jwst/jwst/straylight/tests/test_straylight_step.py:53: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/Users/jdavies/dev/jwst/jwst/stpipe/step.py:385: in run
    step_result = self.process(*args)
/Users/jdavies/dev/jwst/jwst/straylight/straylight_step.py:37: in process
    " 0.1 to 5. It is set to {}".format(self.power))
/Users/jdavies/dev/jwst/jwst/associations/lib/log_config.py:168: in log
    self._log(level, line, args, **kwargs)
/Users/jdavies/miniconda3/envs/jwst_yaml/lib/python3.7/logging/__init__.py:1514: in _log
    self.handle(record)
/Users/jdavies/miniconda3/envs/jwst_yaml/lib/python3.7/logging/__init__.py:1524: in handle
    self.callHandlers(record)
/Users/jdavies/miniconda3/envs/jwst_yaml/lib/python3.7/logging/__init__.py:1586: in callHandlers
    hdlr.handle(record)
/Users/jdavies/miniconda3/envs/jwst_yaml/lib/python3.7/logging/__init__.py:894: in handle
    self.emit(record)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <BreakHandler (ERROR)>
record = <LogRecord: stpipe.straylight, 40, /Users/jdavies/dev/jwst/jwst/straylight/straylight_step.py, 37, "The kernel power parameter is outside the reasonable range of 0.1 to 5. It is set to 6">

    def emit(self, record):
>       raise LoggedException(record)
E       jwst.stpipe.log.LoggedException: The kernel power parameter is outside the reasonable range of 0.1 to 5. It is set to 6

/Users/jdavies/dev/jwst/jwst/stpipe/log.py:82: LoggedException
```
This seems to be an interaction between the `log.error` in `StraylightStep`, and the logging configuration in `associations`.  If one runs the straylight tests on their own, it doesn't happen, and it doesn't always happen when all the tests are run together.

My thought that we shouldn't be logging errors if we are handling them?  Perhaps just warn the user?  And we shouldn't be turning logged errors into raised exceptions in our logging configuration?  Thoughts?

**Comment by jdavies-st** [comment]
And it fact, it might be better to have the ConfigObj validator take care of acceptable ranges for the input values instead of doing it in the code.  So instead of

https://github.com/spacetelescope/jwst/blob/fe85e1da6e5bb7eb606ec3f5dac612ad5e32135d/jwst/straylight/straylight_step.py#L35-L41

Do the following:
```
    spec = """
         method = option('Nearest','ModShepard',default='ModShepard') #Algorithm method
         roi = integer(default=50) # Region of interest given as even integer
         power = float(0.1, 5, default=1.0) # Power of weighting function
```
That said, the validator only gets run on instantiation.  One could then change it to 6 after instantiation, and that would not trigger the validator.  =/

**Comment by jdavies-st** [comment]
Just a note here that the `log.error` in straylight has now been removed, but the issue of `log.error` calls being turned into a full-blown `Exception` via the logging config is still an issue.

**Comment by jdavies-st** [comment]
Resolved by #4510
